kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterService.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.BulkDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyspec.Clusters;
+
+/**
+ * The Kube {@code Service} for a single virtual cluster.
+ * This is named like {@code ${cluster.name}-cluster-${KafkaProxy.metadata.name}}.
+ */
+@KubernetesDependent
+public class ClusterService
+        extends CRUDKubernetesDependentResource<Service, KafkaProxy>
+        implements BulkDependentResource<Service, KafkaProxy> {
+
+    public static final String CLUSTER_INFIX = "-cluster-";
+
+    public ClusterService() {
+        super(Service.class);
+    }
+
+    /**
+     * @return The {@code metadata.name} of the desired {@code Service}.
+     */
+    static String serviceName(KafkaProxy primary, Clusters cluster) {
+        Objects.requireNonNull(primary);
+        Objects.requireNonNull(cluster);
+        String serviceName = cluster.getName() + CLUSTER_INFIX + primary.getMetadata().getName();
```

Comment:
Do we want to include the proxy name in there at all? A consequence of doing that is it couples the DNS name (which all the clients have to use) to the proxy instances that the virtual cluster is running in _today_. Thus preventing someone from refactoring and splitting out one cluster into its own proxy. That feels like it's going to be pretty restrictive. 

The alternative would be to say that the `Service` name is just the cluster name, which would allow the user to refactor how their clusters are grouped within proxy instances at will. We'd need another way of determining which Services in a Namespace belonged to which Proxy.

Reply:
The conclusion from the call was that we should not include the proxy in the name as we think porting virtual clusters from one proxy instance to another should be seamless change from the client point of view.

The hunch is that we can use labels or an owner ref to tie proxies to clusters.